backend/app/Hooks/DiscordOAuthHelper.php

```
@@ -281,23 +281,34 @@ public function checkJ4RServerJoins(string $discordId, string $accessToken, Sess
+                try {
+                    $userUuid = $session->getInfo(UserColumns::UUID, false);
+
+                    UserLock::executeWithLock($userUuid, function () use ($session, $joinedServersArray, $totalCoinsEarned) {
+                        // Update user's joined servers list
+                        $session->setInfo(UserColumns::J4R_JOINED_SERVERS, json_encode($joinedServersArray), false);
+
+                        // Add coins to user's balance
+                        $currentCoins = (int) $session->getInfo(UserColumns::CREDITS, false);
+                        $newCoins = $currentCoins + $totalCoinsEarned;
+                        $session->setInfo(UserColumns::CREDITS, (string) $newCoins, false);
+                    });
```

PHPMD (2.15.0)

Avoid using static access to class '\MythicalDash\Chat\User\UserLock' in method 'checkJ4RServerJoins'. (Clean Code Rules)

(StaticAccess)